src/Raven.Server/Documents/Sharding/Handlers/Processors/OngoingTasks/ShardedOngoingTasksHandlerProcessorForGetOngoingTasks.cs

```diff
@@ -27,23 +27,33 @@ protected override Task HandleRemoteNodeAsync(ProxyCommand<OngoingTasksResult> c
 
         protected override RavenCommand<OngoingTasksResult> CreateCommandForNode(string nodeTag)
         {
-            return new GetOngoingTasksInfoCommand();
+            return new GetOngoingTasksInfoCommand(nodeTag);
         }
 
         protected override bool SupportsCurrentNode => false;
     }
 
     internal class GetOngoingTasksInfoCommand : RavenCommand<OngoingTasksResult>
     {
+        private readonly string _nodeTag;
+
+        public GetOngoingTasksInfoCommand(string nodeTag = null)
+        {
+            _nodeTag = nodeTag;
+        }
+
         public override HttpRequestMessage CreateRequest(JsonOperationContext ctx, ServerNode node, out string url)
         {
             url = $"{node.Url}/databases/{node.Database}/tasks";
 
+            if (_nodeTag != null)
+                url += $"?nodeTag={_nodeTag}";
```

Comment:
If I understand this correctly you create here something like `NW$1/tasks?nodeTag=C` which might go to node A of NW$1 and redirected to node C.
instead you should go directly to the C node, no?

Reply:
Yep. Done